Sources/WendyShared/DeviceModels.swift

```diff
@@ -34,9 +34,153 @@ public struct DevicesCollection: Encodable, Sendable {
+    /// The number of unique devices (counting each unique device name once)
     public var deviceCount: Int {
-        return usbDevices.count + ethernetDevices.count + lanDevices.count
+        return uniqueDeviceNames.count
+    }
+
+    /// Normalize device name for grouping similar devices
+    private func normalizeDeviceName(_ name: String) -> String {
+        // Convert to lowercase, remove common prefixes, and normalize separators
+        var normalized = name.lowercased()
+
+        // Remove common prefixes
+        let prefixes = ["wendyos device", "wendy device", "device"]
+        for prefix in prefixes {
+            if normalized.hasPrefix(prefix + " ") {
+                normalized = String(normalized.dropFirst(prefix.count + 1))
+            }
+        }
+
+        // Normalize all separators (spaces, underscores, hyphens) to hyphens
+        normalized = normalized.replacingOccurrences(of: "_", with: "-")
+        normalized = normalized.replacingOccurrences(of: " ", with: "-")
+
+        // Remove any duplicate hyphens that might result
+        while normalized.contains("--") {
+            normalized = normalized.replacingOccurrences(of: "--", with: "-")
+        }
+
+        // Remove extra whitespace and trim hyphens from ends
+        normalized = normalized.trimmingCharacters(in: CharacterSet(charactersIn: "- "))
+
+        return normalized
+    }
+
+    /// Get unique device names across all interfaces (using normalized names)
+    private var uniqueDeviceNames: Set<String> {
+        var names = Set<String>()
+        for device in usbDevices {
+            names.insert(normalizeDeviceName(device.displayName))
+        }
+        for device in ethernetDevices {
+            names.insert(normalizeDeviceName(device.displayName))
+        }
+        for device in lanDevices {
+            names.insert(normalizeDeviceName(device.displayName))
+        }
+        return names
+    }
+
+    /// Groups all devices by their normalized name
+    private func groupedDevices() -> [(name: String, interfaces: [InterfaceInfo])] {
+        // Use normalized names as keys, but track the best display name
+        var deviceGroups: [String: (displayName: String, interfaces: [InterfaceInfo])] = [:]
+
+        // Helper to choose the best display name (prefer shorter, cleaner names)
+        func betterDisplayName(_ name1: String, _ name2: String) -> String {
+            // Prefer names without "WendyOS Device" prefix
+            if name1.hasPrefix("WendyOS Device") && !name2.hasPrefix("WendyOS Device") {
+                return name2
+            }
+            if !name1.hasPrefix("WendyOS Device") && name2.hasPrefix("WendyOS Device") {
+                return name1
+            }
+            // Otherwise prefer shorter names or the first one
+            return name1.count <= name2.count ? name1 : name2
+        }
+
+        // Group USB devices
+        for device in usbDevices {
```

Comment:
I wonder if we get some common serial or vendor product number that we can key on for grouping, that's the same for all interfaces, so grouping is more robust